setup.sh

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+#
+# Doubtfire MacOS Development Install Script
+# =======================
+#
+# Provides commands to install Doubtfire
+#
+
+#
+# OS Detection functions
+#
+isMac() {
+    if [[ `uname` == Darwin ]]; then
+        # 0 = true
+        return 0
+    else 
+        return 1
+    fi
+}
+
+isLinux() {
```

Comment:
Add documentation to each function. It's self explanatory but for consistencies sake.